libs/utils/analysis/status_analysis.py

```diff
@@ -88,6 +88,11 @@ def plotOverutilized(self, axes=None):
         # Compute intervals in which the system is reported to be overutilized
         bands = [(t, df['len'][t], df['overutilized'][t]) for t in df.index]
 
+        # The last event's duration will always be set to NaN, because the time
+        # delta is computed as (next_event time - cur_event time). This fixes
+        # that by 'manually' setting the duration of the last event.
+        bands[-1] = (bands[-1][0], self._trace.x_max - bands[-1][0], bands[-1][2])
```

Comment:
Should not better move this fix into the DataFrame sanitizer, i.e. `Trace::_sanitize_SchedOverutilized`?
This would allow to always get a DF which has already the `len` field properly populated?